Native dependency not inserted in link args when running doctest

I have a dependency which is generated by pkg-config in a custom build script.

The output of the buildscript is:
```
cargo:rustc-flags=-L native=/vol/home/s0828599/stow/portaudio/lib
cargo:rustc-flags=-l portaudio
cargo:rustc-flags=-l asound
cargo:rustc-flags=-l m
cargo:rustc-flags=-l pthread
```

When running `cargo run --example demo` the native dependency (`-L native=/vol/home/s0828599/stow/portaudio/lib`) is correctly inserted:
```
rustc examples/demo.rs --crate-name demo --crate-type bin -g --out-dir /vol/home/s0828599/playground/rust/portaudio-rs/target/examples --emit=dep-info,link -L dependency=/vol/home/s0828599/playground/rust/portaudio-rs/target -L dependency=/vol/home/s0828599/playground/rust/portaudio-rs/target/deps --extern portaudio-sys=/vol/home/s0828599/playground/rust/portaudio-rs/target/deps/libportaudio-sys-7cbaf4ea274a2241.rlib --extern bitflags=/vol/home/s0828599/playground/rust/portaudio-rs/target/deps/libbitflags-57b03d5337bba57b.rlib --extern portaudio=/vol/home/s0828599/playground/rust/portaudio-rs/target/libportaudio-98112834b31c715f.rlib -L native=/vol/home/s0828599/stow/portaudio/lib
```

However, when running `cargo test` it will fail. The 'normal' tests run ok, but the doc tests output an error. As you can see below, the native dep is not inserted here.

Error:
```
error: linking with `cc` failed: exit code: 1
note: cc '-Wl,--as-needed' '-m64' '-L' '/vol/home/s0828599/stow/rust-nightly/lib/rustlib/x86_64-unknown-linux-gnu/lib' '-o' '/tmp/rustdoctest.IUiecnTvoPsU/rust-out' '/tmp/rustdoctest.IUiecnTvoPsU/rust_out.o' '-Wl,--whole-archive' '-lmorestack' '-Wl,--no-whole-archive' '-Wl,--gc-sections' '-pie' '-nodefaultlibs' '/vol/home/s0828599/playground/rust/portaudio-rs/target/libportaudio-98112834b31c715f.rlib' '/vol/home/s0828599/playground/rust/portaudio-rs/target/deps/libbitflags-57b03d5337bba57b.rlib' '/vol/home/s0828599/playground/rust/portaudio-rs/target/deps/libportaudio-sys-7cbaf4ea274a2241.rlib' '-L' '/vol/home/s0828599/stow/rust-nightly/lib/rustlib/x86_64-unknown-linux-gnu/lib' '-llog-4e7c5e5c' '-L' '/vol/home/s0828599/stow/rust-nightly/lib/rustlib/x86_64-unknown-linux-gnu/lib' '-lstd-4e7c5e5c' '-L' '/vol/home/s0828599/playground/rust/portaudio-rs/target' '-L' '/vol/home/s0828599/playground/rust/portaudio-rs/target/deps' '-L' '/vol/home/s0828599/stow/rust-nightly/lib/rustlib/x86_64-unknown-linux-gnu/lib' '-L' '/vol/home/s0828599/playground/rust/portaudio-rs/.rust/lib/x86_64-unknown-linux-gnu' '-L' '/vol/home/s0828599/playground/rust/portaudio-rs/lib/x86_64-unknown-linux-gnu' '-Wl,--whole-archive' '-Wl,-Bstatic' '-Wl,--no-whole-archive' '-Wl,-Bdynamic' '-lportaudio' '-lportaudio' '-lasound' '-lm' '-lpthread' '-ldl' '-lpthread' '-lrt' '-lgcc_s' '-lpthread' '-lc' '-lm' '-lcompiler-rt'
note: /usr/bin/ld: cannot find -lportaudio
/usr/bin/ld: cannot find -lportaudio
collect2: ld returned 1 exit status
```

Compiler flags:
```
rustdoc --test /vol/home/s0828599/playground/rust/portaudio-rs/src/lib.rs --crate-name portaudio -L /vol/home/s0828599/playground/rust/portaudio-rs/target -L /vol/home/s0828599/playground/rust/portaudio-rs/target/deps --extern portaudio-sys=/vol/home/s0828599/playground/rust/portaudio-rs/target/deps/libportaudio-sys-7cbaf4ea274a2241.rlib --extern pkg-config=/vol/home/s0828599/playground/rust/portaudio-rs/target/deps/libpkg-config-be6313c85913bcf0.rlib --extern bitflags=/vol/home/s0828599/playground/rust/portaudio-rs/target/deps/libbitflags-57b03d5337bba57b.rlib --extern portaudio=/vol/home/s0828599/playground/rust/portaudio-rs/target/libportaudio-98112834b31c715f.rlib
```

If I add the `-L native ...` argument then it runs fine.

--- a/src/cargo/ops/cargo_test.rs
+++ b/src/cargo/ops/cargo_test.rs
@@ -45,6 +45,10 @@ pub fn run_tests(manifest_path: &Path,
          })
          .cwd(compile.package.root());
 
+        for native_dep in compile.native_dirs.values() {
+            p.arg("-L").arg(native_dep);
+        }
+
         if test_args.len() > 0 {
             p.arg("--test-args").arg(&test_args.connect(" "));
         }

